braze-inc/braze-growth-shares-ios-demo-app
# Added SharePlay capability to Video IAM

Beefed up the Video IAM to support SharePlay capability - made possible with the help of [Apple's Demo Project](https://developer.apple.com/documentation/avfoundation/media_playback_and_selection/supporting_coordinated_media_playback)

**To Test**
SharePlay is not available on iOS 15.0 but is available on the iOS 15.1 beta

For iOS 15 users who are eligible to start SharePlay (on a FaceTime call), they will see the SharePlay button appear on the Modal IAM to kick off a group session. 

Users on below iOS 14 can still interact with the IAM but the SharePlay button will not be visible. 

**How it works (high-level)**
- 1st user receives IAM
- 1st user starts SharePlay
- 2nd user taps "Open SharePlay" button
- `PlaybackCoordinator` manages group session for 2nd user on app launch and triggers custom even to launch the video IAM
- SharePlay begins


**Pausing playback**
![image5](https://user-images.githubusercontent.com/5905170/135328974-b1786844-bf05-4004-ac8c-61d489088aa3.png)
![image2](https://user-images.githubusercontent.com/5905170/135328976-d58d658c-43de-4f77-96b9-5df86fdc094a.png)

**iOS 14 users**
![Simulator Screen Shot - iPhone 12 Pro Max - 2021-09-29 at 14 35 21](https://user-images.githubusercontent.com/5905170/135329029-f5a57362-e60d-4add-aa0d-e0a099a19e65.png)
